sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/SourceGenerationContext.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Monitor.OpenTelemetry.Exporter.Internals.Statsbeat;
+using System.Text.Json.Serialization;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals;
+
+#if NET6_0_OR_GREATER
+// "Source Generation" is a mitigation for AOT IL3050:RequiresDynamicCode and IL2026:RequiresUnreferencedCode
+// https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation
```

Comment:
Using source generator should be beneficial regardless of AOT as well - it improves startup times by avoiding costly reflection calls.

Reply:
updated comment